spells.yml

```diff
@@ -9860,6 +9882,32 @@ wall_of_fire:
     - fire
     - sorcerer
     - wizard
+wall_of_flesh:
+  casting_time: 1 standard action
+  components: V, S, M
+  concentration: true
+  description: |-
+    You create a wall of living flesh within range. The wall is 1 foot thick and is composed of ten 10-foot-by-10-foot panels. Each panel must be contiguous with at least one other panel, be vertically oriented, and be supported underneath by a solid surface or on any sides by at least two other panels. If the wall cuts through a creature's space when it appears, the creature is pushed to one side of the wall (your choice).
+
+    Each panel has 10 AC and 30 hit points. A panel is destroyed if it is reduced to 0 hit points, potentially causing other panels to collapse and be destroyed.
+
+    At the start of each of your turns, you can cause one or more wall panels to regenerate a total of 30 hit points. Alternatively, you can cause a new panel to grow from a panel that has all its hit points. The new panel must be connected to the old panel but otherwise follows the rules of panel creation as above.
+
+    The wall has eyes and darkvision out to a range of 30 feet. When a creature the wall can see starts its turn within 5 feet of the wall, the wall attacks with its fleshy tendrils. Make a melee spell attack against the target. On a hit, the target takes 2d8 bludgeoning damage, and if the target is Large or smaller, it is grappled by the wall (escape DC equals your spell save DC).
```

Comment:
Dumb question, but if the wall has a size of, say, Large, and it grapples a creature, and the creature then has enlarge cast on it, making it huge, would it then be able to drag the wall while moving at half speed?

Reply:
RAW no, for a very stupid reason: "When you move, you drag the grappling **creature** with you."

But I'll probably change that to "drag the grappler with you", so I'll also make the wall explicitly immovable (also preventing pushes from eg thunderwave)